src/payment/bolt12.rs

```diff
@@ -264,4 +267,62 @@ impl Bolt12Payment {
 
 		Ok(offer)
 	}
+
+	/// Requests a refund payment for the given [`Refund`].
+	///
+	/// The returned [`Bolt12Invoice`] is for informational purposes only (i.e., isn't needed to
+	/// retrieve the refund).
+	pub fn request_refund(&self, refund: &Refund) -> Result<Bolt12Invoice, Error> {
+		self.channel_manager.request_refund_payment(refund).map_err(|e| {
+			log_error!(self.logger, "Failed to request refund payment: {:?}", e);
+			Error::InvalidRefund
```

Comment:
Similarly here I'm not sure if we want to differentiate when we fail to create the invoice / reply path.